src/libraries/Common/src/System/Net/Http/aspnetcore/Http2/Hpack/Huffman.cs

```diff
@@ -269,36 +273,126 @@ internal class Huffman
-            (28, new[] { 2, 3, 4, 5, 6, 7, 8, 11, 12, 14, 15, 16, 17, 18, 19, 20, 21, 23, 24, 25, 26, 27, 28, 29, 30, 31, 127, 220, 249 }),
-            (30, new[] { 10, 13, 22, 256 })
-        };
-
         public static (uint encoded, int bitLength) Encode(int data)
         {
             return _encodingTable[data];
         }
 
+        private static ushort[] GenerateDecodingLookupTree()
+        {
+            // Decoding lookup tree is a tree of 8 bit lookup tables stored in
+            // one dimensional array of ushort to reduce allocations.
+            // First 256 ushort is lookup table with index 0, next 256 ushort is lookup table with index 1, etc...
+            // loopk_value = [(lookup_table_index << 8) + lookup_index]
+
+            // lookup_index is next 8 bits of huffman code, if there is less than 8 bits in source.
+            // lookup_index MUST be aligned to 8 bits with LSB bits set to anything (zeros are recommended).
+
+            // Lookup value is encoded in ushort as either.
+            // -----------------------------------------------------------------
+            //  15  14  13  12  11  10   9   8   7   6   5   4   3   2   1   0
+            // +---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+
+            // | 0 |   next_lookup_table_index |            not_used           |
+            // +---+---------------------------+-------------------------------+
+            // or
+            // +---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+
+            // | 1 |     number_of_used_bits   |              octet            |
+            // +---+---------------------------+-------------------------------+
+
+            // Bit 15 set indicates a leaf value of decoding tree.
+            // For example value 0x8241 means that we have reached end of huffman code
+            // with result byte 0x41 'A' and from lookup bits count only rightmost 2 bites was used
+            // and rest of bits are part of next huffman code.
+
+            // Bit 15 unset indicates that code is not yet decoded and next lookup table index shall be used
+            // for next n bits of huffman code.
+            // 0 in 'next lookup table index' is considered as decoding error - invalid huffman code
+
+            // Because HPack uses static huffman code defined in RFC https://httpwg.org/specs/rfc7541.html#huffman.code
+            // it is guaranteed that for this huffman code generated decoding lookup tree MUST consist of 15 lookup tables
+            var lut = new ushort[15 * 256];
+
+            int allocatedLookupTableIndex = 0;
+            // Create traverse path for all 0..256 octets, 256 is EOS, see: http://httpwg.org/specs/rfc7541.html#rfc.section.5.2
+            for (int octet = 0; octet <= 256; octet++)
+            {
+                (uint code, int bitsLeft) = Encode(octet);
```

Comment:
I also frown on it a little when I was writing it. I think this will be more readable:
```C#
(uint code, int bitLength) = Encode(octet);
int bitsLeft = bitlenght;
```